packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,316 @@
+  }
+
+  public getContentOfResultTemplate(
+    resultOrFolded: Result | FoldedResult
+  ): HTMLElement | DocumentFragment {
+    const result = (resultOrFolded as FoldedResult).result || resultOrFolded;
+    return this.render ? this.render(result) : this.getTemplate(result)!;
+  }
+
+  private makeDefaultTemplate(): ResultTemplate<DocumentFragment> {
+    const content = document.createDocumentFragment();
+    const linkEl = document.createElement('atomic-result-link');
+    content.appendChild(linkEl);
+    return {
+      content,
+      conditions: [],
+    };
+  }
+
+  public async registerResultTemplates(
+    elements: NodeListOf<TemplateElement>,
+    includeDefaultTemplate = true,
+    onReady: () => void,
+    onError: () => void
+  ) {
+    this.resultTemplatesManager = buildResultTemplatesManager(
+      this.bindings.engine
+    );
+
+    const customTemplates = await Promise.all(
+      Array.from(elements).map(async (resultTemplateElement) => {
+        const template = await resultTemplateElement.getTemplate();
+        if (!template) {
+          onError();
+        }
+        return template;
+      })
+    );
+    const templates = (
+      includeDefaultTemplate ? [this.makeDefaultTemplate()] : []
+    ).concat(
+      customTemplates.filter(identity) as ResultTemplate<DocumentFragment>[]
+    );
+
+    this.resultTemplatesManager.registerTemplates(...templates);
+    onReady();
+  }
+
+  public getResultId(
+    result: Result | FoldedCollection,
+    resultListState: FoldedResultListState | ResultListState
+  ) {
+    return (
+      this.getUnfoldedResult(result).uniqueId + resultListState.searchResponseId
+    );
+  }
+
+  public getUnfoldedResult(result: Result | FoldedResult): Result {
+    return (result as FoldedResult).result ?? result;
+  }
+
+  public handleInfiniteScroll(
+    isEnabled: boolean,
+    host: HTMLElement,
+    resultList: ResultList | FoldedResultList
+  ) {
+    if (!isEnabled) {
+      return;
+    }
+
+    const hasReachedEndOfElement =
+      window.innerHeight + window.scrollY >= host.offsetHeight;
+
+    if (hasReachedEndOfElement) {
+      resultList.fetchMoreResults();
+    }
+  }
+
+  public componentDidRender(
+    resultListState: FoldedResultListState | ResultListState,
+    listWrapperRef?: HTMLDivElement
+  ) {
+    if (resultListState.firstSearchExecuted) {
+      listWrapperRef?.classList.remove('placeholder');
+    }
+  }
+
+  private getClasses({
+    display = 'list',
+    density,
+    imageSize,
+    image,
+    resultListState,
+    resultList,
+  }: AtomicResultListBaseComponent): string {
+    const classes = getResultDisplayClasses(
+      display,
+      density,
+      (imageSize ?? image)!
+    );
+    if (resultListState.firstSearchExecuted && resultList.state.isLoading) {
+      classes.push('loading');
+    }
+    return classes.join(' ');
+  }
+
+  public renderList(props: AtomicResultListBaseComponent) {
```

Comment:
do you mean, instead of doing `this.resultListCommon.renderList(this);`, do `this.resultListCommon.renderList({...anObjectWithEverything);`?
I don't get how we could take a `ResultListCommonProps` if it doesn't have any of the stuff we need for rendering 🤔  I think I need a bit more help

Reply:
perhaps I could `Pick<AtomicResultListBaseComponent, '....'>` ?

Reply:
I basically mean removing `AtomicResultListBaseComponent` altogether, and calling this function like:
```ts
return this.resultListCommon.renderList({
    templateHasError: this.templateHasError,
    listWrapperRef: this.listWrapperRef,
    /* etc */
});
```

Reply:
ah, ok I see. but then, in practice, how do I declare the parameters of `renderList` ? do I then need a new `Props` interface? and then do I copy and paste, like
```ts
interface RenderListProps {
  resultListState: FoldedResultListState | ResultListState;
  resultListCommon: ResultListCommon;
  host: HTMLElement;
  etc...
}
```
even if we have most of that stuff in other interfaces already, like
```ts
export interface ResultsProps extends DisplayOptions {
  resultListState: FoldedResultListState | ResultListState;
  resultListCommon: ResultListCommon;
  bindings: Bindings;
  host: HTMLElement;
  getContentOfResultTemplate(
    result: Result | FoldedResult
  ): HTMLElement | DocumentFragment;

  classes: string;
}

interface DisplayOptions {
  density: ResultDisplayDensity;
  imageSize?: ResultDisplayImageSize;
  image?: ResultDisplayImageSize;
  display?: ResultDisplayLayout;
}
```

Reply:
yes, I think a new prop interface would be good. Basically, I want to eliminate `AtomicResultListBaseComponent` and I don't want to `implement` an interface in addition to initializing `ResultListCommon`.

Reply:
alright, I think I got it!